src/source/avro.rs

```diff
@@ -134,47 +127,65 @@ impl TCPStreamHandler for AvroStreamHandler {
                             streaming = false;
                             break;
                         }
-                        _stream_token => match self.handle_avro_payload(
-                            chans.clone(),
-                            tags,
-                            &mut reader,
-                        ) {
-                            Ok(maybe_id) => {
-                                trace!("Avro payloads processed successfully.");
-                                AVRO_PAYLOAD_SUCCESS_SUM
-                                    .fetch_add(1, Ordering::Relaxed);
-
-                                if maybe_id.is_some() {
-                                    let mut resp = Cursor::new(Vec::new());
-                                    resp.write_u64::<BigEndian>(maybe_id.unwrap())
-                                        .expect("Failed to write response id!");
-                                    stream
-                                        .write_all(resp.get_ref())
-                                        .expect("Failed to write response!");
+                        _stream_token => {
+                            while streaming {
+                                match reader.read() {
+                                    Ok(payload) => {
+                                        let handle_res = self.handle_avro_payload(chans.clone(),tags, payload);
+                                        if handle_res.is_err() {
+                                            streaming = false;
```

Comment:
This probably calls for new internal telemetry

Reply:
Teased apart AVRO_PAYLOAD_PARSE_FAILURE_SUM from AVRO_PAYLOAD_IO_FAILURE_SUM.